gnuradio-runtime/python/gnuradio/gr/bindings/block_gateway_python.cc

```diff
@@ -7,17 +7,200 @@
  *
  */
 
+#include <gnuradio/api.h>
+#include <gnuradio/block.h>
+#include <gnuradio/io_signature.h>
+#include <string>
+
 #include <pybind11/complex.h>
 #include <pybind11/pybind11.h>
 #include <pybind11/stl.h>
 
 namespace py = pybind11;
 
-#include <gnuradio/block_gateway.h>
+
+/*!
+ * The gateway block which performs all the magic.
+ *
+ * The gateway provides access to all the gr::block routines.
+ */
+typedef enum {
+    GW_BLOCK_GENERAL = 0,
+    GW_BLOCK_SYNC,
+    GW_BLOCK_DECIM,
+    GW_BLOCK_INTERP
+} gw_block_t;
+
+class block_gateway : virtual public gr::block
```

Comment:
This is now an implementation detail; no need for virtual inheritance
```suggestion
class block_gateway : public gr::block
```
(unless there's something in the pybind implementation that still needs it?)